spring-rabbit/src/test/java/org/springframework/amqp/rabbit/config/RabbitListenerContainerTestFactory.java

```diff
@@ -27,22 +29,27 @@
 /**
  *
  * @author Stephane Nicoll
+ * @author Gary Russell
  */
 public class RabbitListenerContainerTestFactory implements RabbitListenerContainerFactory<MessageListenerTestContainer> {
 
 	private static final AtomicInteger counter = new AtomicInteger();
 
-	private final List<MessageListenerTestContainer> listenerContainers =
-			new ArrayList<MessageListenerTestContainer>();
+	private final Map<String, MessageListenerTestContainer> listenerContainers =
+			new LinkedHashMap<String, MessageListenerTestContainer>();
 
 	public List<MessageListenerTestContainer> getListenerContainers() {
-		return listenerContainers;
+		return new ArrayList<MessageListenerTestContainer>(this.listenerContainers.values());
+	}
+
+	public MessageListenerTestContainer getListenerContainer(String id) {
+		return this.listenerContainers.get(id);
 	}
 
 	@Override
 	public MessageListenerTestContainer createListenerContainer(RabbitListenerEndpoint endpoint) {
 		MessageListenerTestContainer container = new MessageListenerTestContainer(endpoint);
-		this.listenerContainers.add(container);
+		this.listenerContainers.put(endpoint.getId(), container);
 
 		// resolve the id
 		if (endpoint.getId() == null && endpoint instanceof AbstractRabbitListenerEndpoint) {
```

Comment:
Looks like the previous line of code should be moved after this `if`, otherwise we end up with the `null` key in the map.